v03_pipeline/lib/misc/validation.py

```diff
@@ -56,6 +56,30 @@ def validate_expected_contig_frequency(
             raise SeqrValidationError(msg)
 
 
+def validate_imputed_sex_ploidy(
+    mt: hl.MatrixTable,
+    sex_check_ht: hl.Table,
+) -> None:
+    mt = mt.select_cols(
+        discrepant=(
+            (
+                # All calls are diploid but the sex is Male/Unknown
+                hl.agg.all(mt.GT.is_diploid())
+                & (sex_check_ht[mt.s].predicted_sex != Sex.FEMALE.value)
+            )
+            | (
+                # At least one call is haploid but the sex is Female/Unknown
+                hl.agg.any(~mt.GT.is_diploid())
+                & (sex_check_ht[mt.s].predicted_sex != Sex.MALE.value)
+            )
+        ),
+    )
+    discrepant_rate = mt.aggregate_cols(hl.agg.fraction(mt.discrepant))
+    if discrepant_rate:
+        msg = f'{discrepant_rate:.2%} of samples have misaligned ploidy with their provided imputed sex.  Contact the methods team to investigate.'
+        raise SeqrValidationError(msg)
```

Comment:
hmm questions: 
- what exactly does the failure mode look like here? it would totally fail the loading during the validation step, yeah? 
- what would the methods team investigating this look like ? would we exclude this family from loading by editing what all is subsetted and/or removing it from the airtable? trying to take a step back

Reply:
yeah.. the failure mode is actually quite different for this.... it'll totally fail loading.  These "validation" errors occur at the very entrance to the pipeline, when we're trying to import the callset from VCF.  

I just pulled the exception message from [here](https://github.com/broadinstitute/seqr-loading-pipelines/pull/780/files/772e2b7b3aebd99964ab8b979f8fe8e46df54e83#diff-bfb21e2b0d4e857967483c2ef096c5814dc2644bb9aea28496af70c52dd7fe37R19), so maybe not the clearest... but to me, if there's an issue with the ploidy that points to an issue upstream that we'd need to at least email our DSP friends about.

Reply:
that's fair - and honestly if our observed ploidy disagrees with their metadata for that exact thing, then I could see that deserving a full stop and reporting to our DSP friends:) 

what would the fix be for this, though? Either on their end or ours?